tensorflow_addons/layers/spatial_pyramid_pooling.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Spatial Pyramid Pooling layers"""
+
+import tensorflow as tf
+from tensorflow_addons.layers.adaptive_pooling import AdaptiveAveragePooling2D
+import tensorflow_addons.utils.keras_utils as conv_utils
+
+from typeguard import typechecked
+from typing import Union, Iterable
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class SpatialPyramidPooling2D(tf.keras.layers.Layer):
+    """Performs Spatial Pyramid Pooling.
+
+    Original Paper: https://arxiv.org/pdf/1406.4729.pdf
+
+    Spatial Pyramid Pooling generates a fixed-length representation
+    regardless of input size/scale. It is typically used before layers
```

Comment:
It is typically used before a layer that expects a fixed input shape, e.g. a `Dense` layer.